watchtower-plugin/src/retrier.rs

```diff
@@ -164,17 +288,55 @@ impl Retrier {
     }
 
     fn set_status(&self, status: RetrierStatus) {
-        *self.status.lock().unwrap() = status;
+        *self.status.lock().unwrap() = status.clone();
+
+        // Add or remove retriers from WTClient based on the RetrierStatus
+        if self.is_running() || self.is_idle() {
+            log::debug!("Adding {} to active retriers", self.tower_id);
+            self.wt_client
+                .lock()
+                .unwrap()
+                .retriers
+                .insert(self.tower_id, status);
+        } else if self.is_stopped() {
+            log::debug!("Removing retrier {} from active retriers", self.tower_id);
+            self.wt_client
+                .lock()
+                .unwrap()
+                .retriers
+                .remove(&self.tower_id);
+        }
```

Comment:
can you put a comment here saying we shouldn't remove on status being `Failed` so not to allow a user to manual retry just yet/until the retrier is removed.